frontend/src/theme/ApiDemoPanel/Curl/index.js

```diff
@@ -0,0 +1,300 @@
+import clsx from 'clsx'
+import codegen from 'postman-code-generators'
+import Highlight, { defaultProps } from 'prism-react-renderer'
+import { useTypedSelector } from '@theme/ApiDemoPanel/hooks'
+import buildPostmanRequest from '@theme/ApiDemoPanel/buildPostmanRequest'
+import FloatingButton from '@theme/ApiDemoPanel/FloatingButton'
+import styles from 'docusaurus-theme-openapi/lib/theme/ApiDemoPanel/Curl/styles.module.css'
+
+const languageSet = [
+  {
+    tabName: 'cURL',
+    highlight: 'bash',
+    language: 'curl',
+    variant: 'curl',
+    options: {
+      longFormat: false,
+      followRedirect: true,
+      trimRequestBody: true,
+    },
+  },
+  {
+    tabName: 'Node',
+    highlight: 'javascript',
+    language: 'nodejs',
+    variant: 'axios',
+    options: {
+      ES6_enabled: true,
+      followRedirect: true,
+      trimRequestBody: true,
+    },
+  },
+  {
+    tabName: 'Go',
+    highlight: 'go',
+    language: 'go',
+    variant: 'native',
+    options: {
+      followRedirect: true,
+      trimRequestBody: true,
+    },
+  },
+  {
+    tabName: 'Python',
+    highlight: 'python',
+    language: 'python',
+    variant: 'requests',
+    options: {
+      followRedirect: true,
+      trimRequestBody: true,
+    },
+  },
+]
+const languageTheme = {
+  plain: {
+    color: 'var(--ifm-code-color)',
+  },
+  styles: [
+    {
+      types: ['inserted', 'attr-name'],
+      style: {
+        color: 'var(--openapi-code-green)',
+      },
+    },
+    {
+      types: ['string', 'url'],
+      style: {
+        color: 'var(--openapi-code-green)',
+      },
+    },
+    {
+      types: ['builtin', 'char', 'constant', 'function'],
+      style: {
+        color: 'var(--openapi-code-blue)',
+      },
+    },
+    {
+      types: ['punctuation', 'operator'],
+      style: {
+        color: 'var(--openapi-code-dim)',
+      },
+    },
+    {
+      types: ['class-name'],
+      style: {
+        color: 'var(--openapi-code-orange)',
+      },
+    },
+    {
+      types: ['tag', 'arrow', 'keyword'],
+      style: {
+        color: 'var(--openapi-code-purple)',
+      },
+    },
+    {
+      types: ['boolean'],
+      style: {
+        color: 'var(--openapi-code-red)',
+      },
+    },
+  ],
+}
+
+function getBaseUrl() {
+  /*
+  This is a special case for production.
+
+  We want to be able to build the front end with no value set for
+  `BASE_URL` so that staging, prod and self hosting users
+  can all use the same docker image.
+
+  When deployed to staging, we want the frontend on
+  https://staging.shields.io/ to generate badges with the base
+  https://staging.shields.io/
+  (and we want similar behaviour for users hosting their own instance)
+
+  When we promote to production we want https://shields.io/ and
+  https://www.shields.io/ to both generate badges with the base
+  https://img.shields.io/
+
+  For local dev, we can deal with setting the api and front-end
+  being on different ports using the BASE_URL env var
```

Comment:
I know we already employ this so consider this a tangential future topic, but with our current hosting setup, I wonder if we've got access to do the www -> non-www redirect at some proxy/traffic manager level (e.g. if Fly has something like nginx or haproxy in there in a way that we could configure/use to add the redirect

Reply:
I think the current answer to this is "no"  https://community.fly.io/t/how-to-redirect-from-non-www-to-www/5795